Buttplug.Server/Bluetooth/Devices/LiBo.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Buttplug.Core;
+using Buttplug.Core.Messages;
+
+namespace Buttplug.Server.Bluetooth.Devices
+{
+    internal class LiBoBluetoothInfo : IBluetoothDeviceInfo
+    {
+        public enum Chrs : uint
+        {
+            Tx1 = 0,
+            Tx2,
+            Rx,
+        }
+
+        public Guid[] Services { get; } =
+        {
+            new Guid("00006000-0000-1000-8000-00805f9b34fb"), // Write Service
+            new Guid("00006050-0000-1000-8000-00805f9b34fb"), // Read service (battery)
+
+            // I'm pretty sure that the 2nd one here is ignored due to a call to Services.First() in the bluetooth manager
+        };
+
+        public string[] Names { get; } =
+        {
+            "PiPiJing"
+        };
+
+        public Guid[] Characteristics { get; } =
+        {
+            // tx1 characteristic
+            new Guid("00006001-0000-1000-8000-00805f9b34fb"), // Shock
+
+            // tx2 characteristic
+            new Guid("00006002-0000-1000-8000-00805f9b34fb"), // VibeMode
+
+            // rx characteristic
+            new Guid("00006051-0000-1000-8000-00805f9b34fb"), // Read for battery level
+        };
+
+        public IButtplugDevice CreateDevice(IButtplugLogManager aLogManager,
+            IBluetoothDeviceInterface aInterface)
+        {
+            return new LiBo(aLogManager, aInterface, this);
+        }
+    }
+
+    internal class LiBo : ButtplugBluetoothDevice
+    {
+        private readonly uint _vibratorCount = 1;
+        private readonly uint _shockerCount = 1;
+        private readonly double[] _vibratorSpeed = { 0 };
+        private readonly double[] _shockerIntensity = { 0 };
+
+        public LiBo(IButtplugLogManager aLogManager,
+                      IBluetoothDeviceInterface aInterface,
+                      IBluetoothDeviceInfo aInfo)
+            : base(aLogManager,
+                   $"LiBo Device ({aInterface.Name})",
+                   aInterface,
+                   aInfo)
+        {
+            MsgFuncs.Add(typeof(SingleMotorVibrateCmd), new ButtplugDeviceWrapper(HandleSingleMotorVibrateCmd));
+            MsgFuncs.Add(typeof(VibrateCmd), new ButtplugDeviceWrapper(HandleVibrateCmd, new MessageAttributes() { FeatureCount = _vibratorCount }));
+            // Here goes a handler for Estim shocking
```

Comment:
Make this comment a TODO: so we catch it in searches later. Should also file a followup bug on it.